packages/app-mobile/components/screens/ConfigScreen/NoteExportSection/NoteImportButton.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { _ } from '@joplin/lib/locale';
+import Logger from '@joplin/utils/Logger';
+import { FunctionComponent } from 'react';
+import { join } from 'path';
+import { ConfigScreenStyles } from '../configScreenStyles';
+import InteropService from '@joplin/lib/services/interop/InteropService';
+import pickDocument from '../../../../utils/pickDocument';
+import makeImportExportCacheDirectory from './utils/makeImportExportCacheDirectory';
+import shim from '@joplin/lib/shim';
+import TaskButton, { OnProgressCallback, SetAfterCompleteListenerCallback, TaskStatus } from './TaskButton';
+import { Platform } from 'react-native';
+
+const logger = Logger.create('NoteImportButton');
+
+interface Props {
+	styles: ConfigScreenStyles;
+}
+
+// Exported for search filtering
+export const importButtonDefaultTitle = () => _('Import from JEX');
+export const importButtonDescription = () => _('Import notes from a JEX (Joplin Export) file.');
+
+const getTitle = (taskStatus: TaskStatus) => {
+	if (taskStatus === TaskStatus.InProgress) {
+		return _('Importing...');
+	} else {
+		return importButtonDefaultTitle();
+	}
+};
+
+const runImportTask = async (
+	_onProgress: OnProgressCallback,
+	setAfterCompleteListener: SetAfterCompleteListenerCallback,
+) => {
+	const importTargetPath = join(await makeImportExportCacheDirectory(), 'to-import.jex');
+	logger.info('Importing...');
+
+	setAfterCompleteListener(async (_success: boolean) => {
+		await shim.fsDriver().remove(importTargetPath);
+	});
+
+	const importFiles = await pickDocument(false);
+	if (importFiles.length === 0) {
+		logger.info('Canceled.');
+		return { success: false, warnings: [] };
+	}
+
+	const sourceFileUri = importFiles[0].uri;
+	const sourceFilePath = Platform.select({
+		android: sourceFileUri,
+		ios: decodeURI(sourceFileUri),
+	});
```

Comment:
Currently, we have a similar `Platform.select` in most places we use `pickDocument`. It may make sense to move this into the implementation of `pickDocument`.

Reply:
That's a good idea, would you mind doing this refactoring? Maybe with some wrapper over `pickDocument`

Reply:
`Note.tsx` currently relies on `pickDocument` producing the same output as the image picker library:
https://github.com/laurent22/joplin/blob/ce672915dabc42550550d2eba4df494e02240042/packages/app-mobile/components/screens/Note.tsx#L764

 Doing this refactor would likely cause this assumption to break. Refactoring `pickDocument` should be done while also refactoring `Note.tsx`.